tt_metal/hw/firmware/src/brisck.cc

```diff
@@ -26,11 +26,13 @@ uint8_t noc_index = NOC_INDEX;
 
 void kernel_launch() {
 
+#if !defined(DEBUG_NULL_KERNELS) || defined(DISPATCH_KERNEL)
     firmware_kernel_common_init((void tt_l1_ptr *)MEM_BRISC_INIT_LOCAL_L1_BASE);
 
     noc_local_state_init(noc_index);
 
     kernel_profiler::mark_time(CC_KERNEL_MAIN_START);
     kernel_main();
     kernel_profiler::mark_time(CC_KERNEL_MAIN_END);
+#endif
```

Comment:
Can we clean this up to pass in defines only to worker programs for DEBUG_NULL_KERNELS and not send to dispatch core? That way we don't need extra DISPATCH_KERNEL define.

Reply:
I don't know of a mechanism that informs the build of the difference betwen a worker program and the dispatch programs.  Seems that would be uglier than this from 10K feet.

Reply:
could rename "DISPATCH_KERNEL" to "INFRA_KERNEL" (or some such) to make the mechanism more generic

Reply:
Given that you made a directed test, maybe this ifdef can be commited via a separate PR or at the very least it's own commit referencing the new skip kernel debug mode issue.